### PR TITLE
CON-2419 migrate payment priority to commitments v2

### PR DESCRIPTION
**Pull request by ChrisJWoodcock**

### CODE REVIEW

**Review by hkf-tech — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/EmployerManageApprenticesController.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+using SFA.DAS.EmployerCommitmentsV2.Web.Models.EmployerManageApprentices;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.EmployerCommitmentsV2.Web.Controllers
+{
+    [DasAuthorize(EmployerUserRole.OwnerOrTransactor)]
+    [Route("accounts/{accountHashedId}/apprentices/manage")]
+    public class EmployerManageApprenticesController : Controller
```
[the remark this answers is not in the document]
Comment (ChrisJWoodcock) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/EmployerManageApprenticesController.cs]: Changed to PaymentOrderController - ApprenticeController is getting busy and doesn't really fit with payment order which about providers.

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/EmployerManageApprenticesController.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        }
+
+        [Route("paymentorder")]
+        [HttpPost]
+        public async Task<IActionResult> PaymentOrder([FromServices] IAuthenticationService authenticationService, PaymentOrderViewModel viewModel)
+        {
+            try
+            {
+                await _commitmentsApiClient.UpdateProviderPaymentsPriority(viewModel.AccountId,
+                        new UpdateProviderPaymentsPriorityRequest
```
[the remark this answers is not in the document]
Comment (ChrisJWoodcock) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Controllers/EmployerManageApprenticesController.cs]: Done

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Home/Index.cshtml [hunk 3]
```diff
@@ -28,7 +28,7 @@
             @if (Model.ShowSetPaymentOrderLink)
             {
                 <div class="das-card">
-                    <h2 class="govuk-heading-m"><a id="SetPaymentOrder" href="@Url.CommitmentsLink($"/accounts/{Model.AccountHashedId}/apprentices/manage/paymentorder")" class="govuk-link">Set payment order</a></h2>
+                    <h2 class="govuk-heading-m"><a id="SetPaymentOrder" href="@Url.CommitmentsV2Link($"/accounts/{Model.AccountHashedId}/apprentices/manage/paymentorder")" class="govuk-link">Set payment order</a></h2>
```
[the remark this answers is not in the document]
Comment (ChrisJWoodcock) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Home/Index.cshtml]: Done

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Views/EmployerManageApprentices/PaymentOrder.cshtml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                            <li class="govuk-!-margin-bottom-4">
+                                <label class="govuk-visually-hidden" for="providerList">Select a provider to set order of payment</label>
+                                <select asp-for="ProviderPaymentOrder[priority-1]" asp-items="Model.ProviderSelectListItems()" aria-label="Select a provider" class="govuk-select das-select--select2"></select>
+                            </li>
+                        }
+                    </ol>
+                </fieldset>
+            </div>
+            <div class="govuk-button-group">
+                <input type="submit" class="govuk-button" data-module="govuk-button" value="Save and continue" />
```
[the remark this answers is not in the document]
Comment (ChrisJWoodcock) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Views/EmployerManageApprentices/PaymentOrder.cshtml]: Done

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Views/EmployerManageApprentices/PaymentOrder.cshtml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                            <span class="govuk-visually-hidden">Error:</span> @ViewData.ModelState[nameof(Model.ProviderPaymentOrder)].Errors.First().ErrorMessage
+                        </span>
+                    }
+
+                    <ol class="govuk-list govuk-list--number govuk-!-width-two-thirds govuk-!-margin-top-6 govuk-!-margin-bottom-9">
+                        @for (int priority = 1; priority <= Model.ProviderPaymentOrder.Count(); priority++)
+                        {
+                            <li class="govuk-!-margin-bottom-4">
+                                <label class="govuk-visually-hidden" for="providerList">Select a provider to set order of payment</label>
+                                <select asp-for="ProviderPaymentOrder[priority-1]" asp-items="Model.ProviderSelectListItems()" aria-label="Select a provider" class="govuk-select das-select--select2"></select>
```
[the remark this answers is not in the document]
Comment (ChrisJWoodcock) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Views/EmployerManageApprentices/PaymentOrder.cshtml]: Done

**Review by hkf-tech — APPROVED** [review]

**Pull request merged**